### ISSUE TITLE
The recursive directory search doesn't work when mounting ntfs file to wsl.

### ISSUE THREAD

**Issue by chinise** [watchexec/watchexec#693]
Everything works fine when using ext4. Also when wachexec is running in the same directory as the file being changed.

- Watchexec's version
1.23.0

- The OS you're using
wsl 2 on windows 11

- A log with `-vvv --log-file`
```
{"timestamp":"2023-11-26T21:26:56.505845Z","level":"INFO","fields":{"message":"logging initialised"},"target":"watchexec_cli"}
{"timestamp":"2023-11-26T21:26:56.506112Z","level":"DEBUG","fields":{"message":"arguments","args":"Args { command: [\"python3\", \"inside_test/test.py\"], paths: [], screen_clear: None, on_busy_update: Queue, watch_when_idle: false, restart: false, signal: None, kill: false, stop_signal: None, stop_timeout: TimeSpan(60s), debounce: TimeSpan(50ms), stdin_quit: false, no_vcs_ignore: false, no_project_ignore: false, no_global_ignore: false, no_default_ignore: false, no_discover_ignore: false, postpone: false, delay_run: None, poll: None, shell: None, no_shell: false, no_shell_long: false, no_environment: false, emit_events_to: Environment, env: [], no_process_group: false, once: false, notify: false, project_origin: None, workdir: None, filter_extensions: [], filter_patterns: [], filter_files: [], ignore_patterns: [], ignore_files: [], filter_fs_events: [Create, Remove, Rename, Modify, Metadata], filter_fs_meta: false, print_events: false, verbose: Some(3), log_file: Some(\".\"), manual: false, completions: None }"},"target":"watchexec_cli"}
{"timestamp":"2023-11-26T21:26:56.513579Z","level":"INFO","fields":{"message":"constructing Watchexec from CLI","version":"1.23.0"},"target":"watchexec_cli"}
{"timestamp":"2023-11-26T21:26:56.513814Z","level":"DEBUG","fields":{"message":"InitConfig: on_error"},"target":"watchexec::config"}
{"timestamp":"2023-11-26T21:26:56.514265Z","level":"DEBUG","fields":{"message":"new"},"target":"watchexec_cli::config::runtime","span":{"name":"args-runtime"},"spans":[]}
{"timestamp":"2023-11-26T21:26:56.514499Z","level":"DEBUG","fields":{"message":"RuntimeConfig: command","command":"Shell { shell: Unix(\"sh\"), args: [], command: \"python3 inside_test/test.py\" }"},"target":"watchexec::config","span":{"name":"args-runtime"},"spans":[{"name":"args-runtime"}]}
{"timestamp":"2023-11-26T21:26:56.514693Z","level":"DEBUG","fields":{"message":"RuntimeConfig: pathset","pathset":"[WatchedPath(\"/mnt/b/Projekty/Tests/watchexec_test\")]"},"target":"watchexec::config","span":{"name":"args-runtime"},"spans":[{"name":"args-runtime"}]}
{"timestamp":"2023-11-26T21:26:56.514912Z","level":"DEBUG","fields":{"message":"RuntimeConfig: throttle","throttle":"50ms"},"target":"watchexec::config","span":{"name":"args-runtime"},"spans":[{"name":"args-runtime"}]}
{"timestamp":"2023-11-26T21:26:56.515122Z","level":"DEBUG","fields":{"message":"RuntimeConfig: command_grouped","grouped":"true"},"target":"watchexec::config","span":{"name":"args-runtime"},"spans":[{"name":"args-runtime"}]}
{"timestamp":"2023-11-26T21:26:56.515325Z","level":"DEBUG","fields":{"message":"RuntimeConfig: on_action"},"target":"watchexec::config","span":{"name":"args-runtime"},"spans":[{"name":"args-runtime"}]}
{"timestamp":"2023-11-26T21:26:56.515535Z","level":"DEBUG","fields":{"message":"additional environment variables to add to command","add_envs":"{}"},"target":"watchexec_cli::config::runtime","span":{"name":"args-runtime"},"spans":[{"name":"args-runtime"}]}
{"timestamp":"2023-11-26T21:26:56.515800Z","level":"DEBUG","fields":{"message":"RuntimeConfig: on_pre_spawn"},"target":"watchexec::config","span":{"name":"args-runtime"},"spans":[{"name":"args-runtime"}]}
{"timestamp":"2023-11-26T21:26:56.515997Z","level":"DEBUG","fields":{"message":"RuntimeConfig: on_post_spawn"},"target":"watchexec::config","span":{"name":"args-runtime"},"spans":[{"name":"args-runtime"}]}
{"timestamp":"2023-11-26T21:26:56.516170Z","level":"DEBUG","fields":{"message":"close","time.busy":"1.68ms","time.idle":"225µs"},"target":"watchexec_cli::config::runtime","span":{"name":"args-runtime"},"spans":[]}
{"timestamp":"2023-11-26T21:26:56.516487Z","level":"DEBUG","fields":{"message":"current directory","curdir":"\"/mnt/b/Projekty/Tests/watchexec_test\""},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-11-26T21:26:56.516744Z","level":"DEBUG","fields":{"message":"home directory","homedir":"Some(\"/home/maciekwsl\")"},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-11-26T21:26:56.517100Z","level":"DEBUG","fields":{"message":"resolved whether the homedir is explicitly requested","homedir_requested":"false"},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-11-26T21:26:56.517275Z","level":"DEBUG","fields":{"message":"no paths, using current directory"},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-11-26T21:26:56.517437Z","level":"DEBUG","fields":{"message":"resolved all watched paths","paths":"{\"/mnt/b/Projekty/Tests/watchexec_test\"}"},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-11-26T21:26:56.524070Z","level":"DEBUG","fields":{"message":"no origins, using current directory"},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-11-26T21:26:56.524325Z","level":"DEBUG","fields":{"message":"resolved all project origins","origins":"{\"/mnt/b/Projekty/Tests/watchexec_test\"}"},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-11-26T21:26:56.524619Z","level":"INFO","fields":{"message":"resolved common/project origin","project_origin":"\"/mnt/b/Projekty/Tests/watchexec_test\""},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-11-26T21:26:56.524782Z","level":"INFO","fields":{"message":"resolved working directory","workdir":"\"/mnt/b/Projekty/Tests/watchexec_test\""},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-11-26T21:26:56.526285Z","level":"INFO","fields":{"message":"resolved vcs types","vcs_types":"[]"},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-11-26T21:26:56.533458Z","level":"DEBUG","fields":{"message":"discovered ignore files from project origin","ignores":"[]"},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-11-26T21:26:56.534079Z","level":"DEBUG","fields":{"message":"discovered ignore files from environment","global_ignores":"[]"},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-11-26T21:26:56.534232Z","level":"DEBUG","fields":{"message":"combined and applied overall vcs filter over ignores","ignores":"[]","vcs_types":"[]"},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-11-26T21:26:56.534364Z","level":"DEBUG","fields":{"message":"combined with ignore files from command line / env","ignores":"[]","args.ignore_files":"[]"},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-11-26T21:26:56.534560Z","level":"INFO","fields":{"message":"found some ignores","files":"[]"},"target":"watchexec_cli::filterer::common"}
{"timestamp":"2023-11-26T21:26:56.534784Z","level":"INFO","fields":{"message":"initialising Globset filterer"},"target":"watchexec_cli::filterer::globset"}
{"timestamp":"2023-11-26T21:26:56.535158Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/*.py[co]\", re: \"(?-u)^(?:/?|.*/)[^/]*\\\\.py[co]$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, ZeroOrMore, Literal('.'), Literal('p'), Literal('y'), Class { negated: false, ranges: [('c', 'c'), ('o', 'o')] }]) }","log.target":"globset","log.module_path":"globset","log.file":"/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-11-26T21:26:56.535480Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/.#*\", re: \"(?-u)^(?:/?|.*/)\\\\.\\\\#[^/]*$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('.'), Literal('#'), ZeroOrMore]) }","log.target":"globset","log.module_path":"globset","log.file":"/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-11-26T21:26:56.535718Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/.*.sw?\", re: \"(?-u)^(?:/?|.*/)\\\\.[^/]*\\\\.sw[^/]$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('.'), ZeroOrMore, Literal('.'), Literal('s'), Literal('w'), Any]) }","log.target":"globset","log.module_path":"globset","log.file":"/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-11-26T21:26:56.535951Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/.*.sw?x\", re: \"(?-u)^(?:/?|.*/)\\\\.[^/]*\\\\.sw[^/]x$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('.'), ZeroOrMore, Literal('.'), Literal('s'), Literal('w'), Any, Literal('x')]) }","log.target":"globset","log.module_path":"globset","log.file":"/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-11-26T21:26:56.536174Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/.bzr/**/*\", re: \"(?-u)^(?:/?|.*/)\\\\.bzr(?:/|/.*/)[^/]*$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('.'), Literal('b'), Literal('z'), Literal('r'), RecursiveZeroOrMore, ZeroOrMore]) }","log.target":"globset","log.module_path":"globset","log.file":"/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-11-26T21:26:56.536323Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/_darcs/**/*\", re: \"(?-u)^(?:/?|.*/)_darcs(?:/|/.*/)[^/]*$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('_'), Literal('d'), Literal('a'), Literal('r'), Literal('c'), Literal('s'), RecursiveZeroOrMore, ZeroOrMore]) }","log.target":"globset","log.module_path":"globset","log.file":"/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-11-26T21:26:56.536440Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/.fossil-settings/**/*\", re: \"(?-u)^(?:/?|.*/)\\\\.fossil\\\\-settings(?:/|/.*/)[^/]*$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('.'), Literal('f'), Literal('o'), Literal('s'), Literal('s'), Literal('i'), Literal('l'), Literal('-'), Literal('s'), Literal('e'), Literal('t'), Literal('t'), Literal('i'), Literal('n'), Literal('g'), Literal('s'), RecursiveZeroOrMore, ZeroOrMore]) }","log.target":"globset","log.module_path":"globset","log.file":"/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-11-26T21:26:56.536548Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/.git/**/*\", re: \"(?-u)^(?:/?|.*/)\\\\.git(?:/|/.*/)[^/]*$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('.'), Literal('g'), Literal('i'), Literal('t'), RecursiveZeroOrMore, ZeroOrMore]) }","log.target":"globset","log.module_path":"globset","log.file":"/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-11-26T21:26:56.536655Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/.hg/**/*\", re: \"(?-u)^(?:/?|.*/)\\\\.hg(?:/|/.*/)[^/]*$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('.'), Literal('h'), Literal('g'), RecursiveZeroOrMore, ZeroOrMore]) }","log.target":"globset","log.module_path":"globset","log.file":"/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-11-26T21:26:56.536757Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/.pijul/**/*\", re: \"(?-u)^(?:/?|.*/)\\\\.pijul(?:/|/.*/)[^/]*$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('.'), Literal('p'), Literal('i'), Literal('j'), Literal('u'), Literal('l'), RecursiveZeroOrMore, ZeroOrMore]) }","log.target":"globset","log.module_path":"globset","log.file":"/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-11-26T21:26:56.536864Z","level":"DEBUG","fields":{"message":"glob converted to regex: Glob { glob: \"**/.svn/**/*\", re: \"(?-u)^(?:/?|.*/)\\\\.svn(?:/|/.*/)[^/]*$\", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true, empty_alternates: false }, tokens: Tokens([RecursivePrefix, Literal('.'), Literal('s'), Literal('v'), Literal('n'), RecursiveZeroOrMore, ZeroOrMore]) }","log.target":"globset","log.module_path":"globset","log.file":"/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":426},"target":"globset"}
{"timestamp":"2023-11-26T21:26:56.536962Z","level":"DEBUG","fields":{"message":"built glob set; 0 literals, 1 basenames, 0 extensions, 0 prefixes, 0 suffixes, 2 required extensions, 11 regexes","log.target":"globset","log.module_path":"globset","log.file":"/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/globset-0.4.13/src/lib.rs","log.line":431},"target":"globset"}
{"timestamp":"2023-11-26T21:26:56.537437Z","level":"DEBUG","fields":{"message":"globset filterer built","origin":"\"/mnt/b/Projekty/Tests/watchexec_test\"","num_filters":"0","num_neg_filters":"0","num_ignores":"14","num_in_ignore_files":"(0, 0)","num_neg_ignores":"0","num_extensions":"0"},"target":"watchexec_filterer_globset"}
{"timestamp":"2023-11-26T21:26:56.537550Z","level":"DEBUG","fields":{"message":"RuntimeConfig: filterer","filterer":"WatchexecFilterer { inner: GlobsetFilterer { origin: \"/mnt/b/Projekty/Tests/watchexec_test\", filters: Gitignore { set: GlobSet { len: 0, strats: [] }, root: \"/mnt/b/Projekty/Tests/watchexec_test\", globs: [], num_ignores: 0, num_whitelists: 0, matches: Some(ThreadLocal { local_data: None }) }, ignores: Gitignore { set: GlobSet { len: 14, strats: [Extension(ExtensionStrategy({})), BasenameLiteral(BasenameLiteralStrategy({[46, 68, 83, 95, 83, 116, 111, 114, 101]: [0]})), Literal(LiteralStrategy({})), Suffix(SuffixStrategy { matcher: AhoCorasick(dfa::DFA(\nD 000000: \\x00 => 0\nF 000001:\n >000002: \\x00 => 2\n  000003: \\x00 => 0\nmatch kind: Standard\nprefilter: false\nstate length: 4\npattern length: 0\nshortest pattern length: 18446744073709551615\nlongest pattern length: 0\nalphabet length: 1\nstride: 1\nbyte classes: ByteClasses(0 => [0-255])\nmemory usage: 16\n)\n), map: [], longest: 0 }), Prefix(PrefixStrategy { matcher: AhoCorasick(dfa::DFA(\nD 000000: \\x00 => 0\nF 000001:\n >000002: \\x00 => 2\n  000003: \\x00 => 0\nmatch kind: Standard\nprefilter: false\nstate length: 4\npattern length: 0\nshortest pattern length: 18446744073709551615\nlongest pattern length: 0\nalphabet length: 1\nstride: 1\nbyte classes: ByteClasses(0 => [0-255])\nmemory usage: 16\n)\n), map: [], longest: 0 }), RequiredExtension(RequiredExtensionStrategy({[46, 108, 111, 103]: [(1, Regex(\"(?-u)^(?:/?|.*/)watchexec\\\\.[^/]*\\\\.log$\"))], [46, 107, 97, 116, 101, 45, 115, 119, 112]: [(4, Regex(\"(?-u)^(?:/?|.*/)\\\\.[^/]*\\\\.kate\\\\-swp$\"))]})), Regex(RegexSetStrategy { matcher: RegexSet([\"(?-u)^(?:/?|.*/)[^/]*\\\\.py[co]$\", \"(?-u)^(?:/?|.*/)\\\\.\\\\#[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.[^/]*\\\\.sw[^/]$\", \"(?-u)^(?:/?|.*/)\\\\.[^/]*\\\\.sw[^/]x$\", \"(?-u)^(?:/?|.*/)\\\\.bzr(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)_darcs(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.fossil\\\\-settings(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.git(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.hg(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.pijul(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.svn(?:/|/.*/)[^/]*$\"]), map: [2, 3, 5, 6, 7, 8, 9, 10, 11, 12, 13] })] }, root: \"/mnt/b/Projekty/Tests/watchexec_test\", globs: [Glob { from: None, original: \"**/.DS_Store\", actual: \"**/.DS_Store\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"watchexec.*.log\", actual: \"**/watchexec.*.log\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"*.py[co]\", actual: \"**/*.py[co]\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \".#*\", actual: \"**/.#*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \".*.kate-swp\", actual: \"**/.*.kate-swp\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \".*.sw?\", actual: \"**/.*.sw?\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \".*.sw?x\", actual: \"**/.*.sw?x\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.bzr/**\", actual: \"**/.bzr/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/_darcs/**\", actual: \"**/_darcs/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.fossil-settings/**\", actual: \"**/.fossil-settings/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.git/**\", actual: \"**/.git/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.hg/**\", actual: \"**/.hg/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.pijul/**\", actual: \"**/.pijul/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.svn/**\", actual: \"**/.svn/**/*\", is_whitelist: false, is_only_dir: false }], num_ignores: 14, num_whitelists: 0, matches: Some(ThreadLocal { local_data: None }) }, ignore_files: IgnoreFilterer(IgnoreFilter { origin: \"/mnt/b/Projekty/Tests/watchexec_test\", ignores: Trie { length: 1, node: TrieNode { key: NibbleVec [], key_value: None, child_count: 1, children: [None, None, Some(TrieNode { key: NibbleVec [2, 15], key_value: Some(KeyValue { key: \"/\", value: Ignore { gitignore: Gitignore { set: GlobSet { len: 0, strats: [] }, root: \"\", globs: [], num_ignores: 0, num_whitelists: 0, matches: None }, builder: None } }), child_count: 0, children: [None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None] }), None, None, None, None, None, None, None, None, None, None, None, None, None] } } }), extensions: [] }, fs_events: [Create, Remove, Rename, Modify, Metadata] }"},"target":"watchexec::config"}
{"timestamp":"2023-11-26T21:26:56.537733Z","level":"INFO","fields":{"message":"initialising Watchexec runtime"},"target":"watchexec_cli"}
{"timestamp":"2023-11-26T21:26:56.537853Z","level":"DEBUG","fields":{"message":"initialising","init":"InitConfig { error_channel_size: 64, event_channel_size: 1024, .. }","runtime":"RuntimeConfig { fs: WorkingData { pathset: [WatchedPath(\"/mnt/b/Projekty/Tests/watchexec_test\")], watcher: Native }, keyboard: WorkingData { eof: false }, action: WorkingData { throttle: 50ms, commands: [Shell { shell: Unix(\"sh\"), args: [], command: \"python3 inside_test/test.py\" }], grouped: true, filterer: WatchexecFilterer { inner: GlobsetFilterer { origin: \"/mnt/b/Projekty/Tests/watchexec_test\", filters: Gitignore { set: GlobSet { len: 0, strats: [] }, root: \"/mnt/b/Projekty/Tests/watchexec_test\", globs: [], num_ignores: 0, num_whitelists: 0, matches: Some(ThreadLocal { local_data: None }) }, ignores: Gitignore { set: GlobSet { len: 14, strats: [Extension(ExtensionStrategy({})), BasenameLiteral(BasenameLiteralStrategy({[46, 68, 83, 95, 83, 116, 111, 114, 101]: [0]})), Literal(LiteralStrategy({})), Suffix(SuffixStrategy { matcher: AhoCorasick(dfa::DFA(\nD 000000: \\x00 => 0\nF 000001:\n >000002: \\x00 => 2\n  000003: \\x00 => 0\nmatch kind: Standard\nprefilter: false\nstate length: 4\npattern length: 0\nshortest pattern length: 18446744073709551615\nlongest pattern length: 0\nalphabet length: 1\nstride: 1\nbyte classes: ByteClasses(0 => [0-255])\nmemory usage: 16\n)\n), map: [], longest: 0 }), Prefix(PrefixStrategy { matcher: AhoCorasick(dfa::DFA(\nD 000000: \\x00 => 0\nF 000001:\n >000002: \\x00 => 2\n  000003: \\x00 => 0\nmatch kind: Standard\nprefilter: false\nstate length: 4\npattern length: 0\nshortest pattern length: 18446744073709551615\nlongest pattern length: 0\nalphabet length: 1\nstride: 1\nbyte classes: ByteClasses(0 => [0-255])\nmemory usage: 16\n)\n), map: [], longest: 0 }), RequiredExtension(RequiredExtensionStrategy({[46, 108, 111, 103]: [(1, Regex(\"(?-u)^(?:/?|.*/)watchexec\\\\.[^/]*\\\\.log$\"))], [46, 107, 97, 116, 101, 45, 115, 119, 112]: [(4, Regex(\"(?-u)^(?:/?|.*/)\\\\.[^/]*\\\\.kate\\\\-swp$\"))]})), Regex(RegexSetStrategy { matcher: RegexSet([\"(?-u)^(?:/?|.*/)[^/]*\\\\.py[co]$\", \"(?-u)^(?:/?|.*/)\\\\.\\\\#[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.[^/]*\\\\.sw[^/]$\", \"(?-u)^(?:/?|.*/)\\\\.[^/]*\\\\.sw[^/]x$\", \"(?-u)^(?:/?|.*/)\\\\.bzr(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)_darcs(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.fossil\\\\-settings(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.git(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.hg(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.pijul(?:/|/.*/)[^/]*$\", \"(?-u)^(?:/?|.*/)\\\\.svn(?:/|/.*/)[^/]*$\"]), map: [2, 3, 5, 6, 7, 8, 9, 10, 11, 12, 13] })] }, root: \"/mnt/b/Projekty/Tests/watchexec_test\", globs: [Glob { from: None, original: \"**/.DS_Store\", actual: \"**/.DS_Store\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"watchexec.*.log\", actual: \"**/watchexec.*.log\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"*.py[co]\", actual: \"**/*.py[co]\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \".#*\", actual: \"**/.#*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \".*.kate-swp\", actual: \"**/.*.kate-swp\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \".*.sw?\", actual: \"**/.*.sw?\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \".*.sw?x\", actual: \"**/.*.sw?x\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.bzr/**\", actual: \"**/.bzr/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/_darcs/**\", actual: \"**/_darcs/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.fossil-settings/**\", actual: \"**/.fossil-settings/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.git/**\", actual: \"**/.git/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.hg/**\", actual: \"**/.hg/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.pijul/**\", actual: \"**/.pijul/**/*\", is_whitelist: false, is_only_dir: false }, Glob { from: None, original: \"**/.svn/**\", actual: \"**/.svn/**/*\", is_whitelist: false, is_only_dir: false }], num_ignores: 14, num_whitelists: 0, matches: Some(ThreadLocal { local_data: None }) }, ignore_files: IgnoreFilterer(IgnoreFilter { origin: \"/mnt/b/Projekty/Tests/watchexec_test\", ignores: Trie { length: 1, node: TrieNode { key: NibbleVec [], key_value: None, child_count: 1, children: [None, None, Some(TrieNode { key: NibbleVec [2, 15], key_value: Some(KeyValue { key: \"/\", value: Ignore { gitignore: Gitignore { set: GlobSet { len: 0, strats: [] }, root: \"\", globs: [], num_ignores: 0, num_whitelists: 0, matches: None }, builder: None } }), child_count: 0, children: [None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None] }), None, None, None, None, None, None, None, None, None, None, None, None, None] } } }), extensions: [] }, fs_events: [Create, Remove, Rename, Modify, Metadata] }, .. } }","pid":"4526","version":"2.3.0"},"target":"watchexec::watchexec"}
{"timestamp":"2023-11-26T21:26:56.537997Z","level":"DEBUG","fields":{"message":"kicking off with empty event"},"target":"watchexec_cli"}
{"timestamp":"2023-11-26T21:26:56.538122Z","level":"INFO","fields":{"message":"running main loop"},"target":"watchexec_cli"}
{"timestamp":"2023-11-26T21:26:56.538241Z","level":"DEBUG","fields":{"message":"handing over main task handle"},"target":"watchexec::watchexec"}
{"timestamp":"2023-11-26T21:26:56.538281Z","level":"DEBUG","fields":{"message":"starting main task"},"target":"watchexec::watchexec"}
{"timestamp":"2023-11-26T21:26:56.538606Z","level":"DEBUG","fields":{"message":"spawning subtask","subtask":"action"},"target":"watchexec::watchexec"}
{"timestamp":"2023-11-26T21:26:56.538857Z","level":"DEBUG","fields":{"message":"spawning subtask","subtask":"fs"},"target":"watchexec::watchexec"}
{"timestamp":"2023-11-26T21:26:56.539139Z","level":"DEBUG","fields":{"message":"spawning subtask","subtask":"signal"},"target":"watchexec::watchexec"}
{"timestamp":"2023-11-26T21:26:56.539181Z","level":"INFO","fields":{"message":"action constructed","action":"Action { events: [Event { tags: [], metadata: {} }], outcome: OnceCell(Uninit) }"},"target":"watchexec::action::worker"}
{"timestamp":"2023-11-26T21:26:56.539609Z","level":"DEBUG","fields":{"message":"running action handler"},"target":"watchexec::action::worker"}
{"timestamp":"2023-11-26T21:26:56.539424Z","level":"DEBUG","fields":{"message":"launching filesystem worker"},"target":"watchexec::fs"}
{"timestamp":"2023-11-26T21:26:56.540184Z","level":"DEBUG","fields":{"message":"creating new watcher","kind":"Native"},"target":"watchexec::fs"}
{"timestamp":"2023-11-26T21:26:56.539962Z","level":"DEBUG","fields":{"message":"action handler finished","outcome":"IfRunning(Both(Wait, Start), Start)"},"target":"watchexec::action::worker"}
{"timestamp":"2023-11-26T21:26:56.540702Z","level":"INFO","fields":{"message":"outcome resolved","outcome":"Start"},"target":"watchexec::action::worker"}
{"timestamp":"2023-11-26T21:26:56.541005Z","level":"DEBUG","fields":{"message":"spawning outcome worker","outcome":"Start","gen":"1"},"target":"watchexec::action::outcome_worker"}
{"timestamp":"2023-11-26T21:26:56.539386Z","level":"DEBUG","fields":{"message":"spawning subtask","subtask":"keyboard"},"target":"watchexec::watchexec"}
{"timestamp":"2023-11-26T21:26:56.540581Z","level":"DEBUG","fields":{"message":"applying changes to the watcher","to_watch":"[WatchedPath(\"/mnt/b/Projekty/Tests/watchexec_test\")]","to_drop":"[]"},"target":"watchexec::fs"}
{"timestamp":"2023-11-26T21:26:56.541360Z","level":"DEBUG","fields":{"message":"action process done"},"target":"watchexec::action::worker"}
{"timestamp":"2023-11-26T21:26:56.541600Z","level":"DEBUG","fields":{"message":"spawning subtask","subtask":"error_hook"},"target":"watchexec::watchexec"}
{"timestamp":"2023-11-26T21:26:56.541634Z","level":"DEBUG","fields":{"message":"launching unix signal worker"},"target":"watchexec::signal"}
{"timestamp":"2023-11-26T21:26:56.542144Z","level":"DEBUG","fields":{"message":"outcome worker finished","outcome":"Start","gen":"1"},"target":"watchexec::action::outcome_worker"}
{"timestamp":"2023-11-26T21:26:56.542687Z","level":"DEBUG","fields":{"message":"new"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[]}
{"timestamp":"2023-11-26T21:26:56.542816Z","level":"DEBUG","fields":{"message":"preparing command","grouped":"true","command":"Shell { shell: Unix(\"sh\"), args: [], command: \"python3 inside_test/test.py\" }"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-11-26T21:26:56.542988Z","level":"DEBUG","fields":{"message":"running pre-spawn handler"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-11-26T21:26:56.543207Z","level":"INFO","fields":{"message":"spawning command","command":"Command { std: \"sh\" \"-c\" \"python3 inside_test/test.py\", kill_on_drop: false }"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-11-26T21:26:56.543657Z","level":"DEBUG","fields":{"message":"resetting process sigmask","unblocking":"SigSet { sigset: sigset_t { __val: [18446744067267100670, 4096, 140354498072720, 298, 18446744073709550976, 93876999940542, 140355025356304, 0, 140355025356296, 562949953421311, 93877039296456, 18444492273895866368, 0, 93876999421176, 140355025356296, 72479423512712096] } }"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-11-26T21:26:56.543917Z","level":"DEBUG","fields":{"message":"sigmask reset","oldset":"SigSet { sigset: sigset_t { __val: [0, 1, 140354498083776, 93877039295896, 140354498080432, 140354498073136, 274877906944, 0, 140355025356304, 0, 93877039295896, 0, 1, 268435456, 0, 140355025345152] } }"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-11-26T21:26:56.544382Z","level":"INFO","fields":{"message":"process group spawned","pgid":"4535"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-11-26T21:26:56.544594Z","level":"DEBUG","fields":{"message":"running post-spawn handler"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-11-26T21:26:56.544908Z","level":"DEBUG","fields":{"message":"spawned process","process":"Grouped(AsyncGroupChild)","pid":"4535"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-11-26T21:26:56.573022Z","level":"DEBUG","fields":{"message":"creating synthetic process completion event","event":"Event { tags: [Source(Internal), ProcessCompletion(Some(Success))], metadata: {} }"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-11-26T21:26:56.573285Z","level":"DEBUG","fields":{"message":"no more commands to supervise"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[{"name":"supervisor"}]}
{"timestamp":"2023-11-26T21:26:56.573578Z","level":"DEBUG","fields":{"message":"close","time.busy":"2.81ms","time.idle":"28.1ms"},"target":"watchexec::command::supervisor","span":{"name":"supervisor"},"spans":[]}
{"timestamp":"2023-11-26T21:26:56.625014Z","level":"INFO","fields":{"message":"action constructed","action":"Action { events: [Event { tags: [Source(Internal), ProcessCompletion(Some(Success))], metadata: {} }], outcome: OnceCell(Uninit) }"},"target":"watchexec::action::worker"}
{"timestamp":"2023-11-26T21:26:56.625477Z","level":"DEBUG","fields":{"message":"running action handler"},"target":"watchexec::action::worker"}
{"timestamp":"2023-11-26T21:26:56.625908Z","level":"DEBUG","fields":{"message":"action handler finished","outcome":"DoNothing"},"target":"watchexec::action::worker"}
{"timestamp":"2023-11-26T21:26:56.626306Z","level":"INFO","fields":{"message":"outcome resolved","outcome":"DoNothing"},"target":"watchexec::action::worker"}
{"timestamp":"2023-11-26T21:26:56.626641Z","level":"DEBUG","fields":{"message":"spawning outcome worker","outcome":"DoNothing","gen":"2"},"target":"watchexec::action::outcome_worker"}
{"timestamp":"2023-11-26T21:26:56.626952Z","level":"DEBUG","fields":{"message":"action process done"},"target":"watchexec::action::worker"}
{"timestamp":"2023-11-26T21:26:56.627359Z","level":"DEBUG","fields":{"message":"outcome worker finished","outcome":"DoNothing","gen":"2"},"target":"watchexec::action::outcome_worker"}
{"timestamp":"2023-11-26T21:27:46.325286Z","level":"DEBUG","fields":{"message":"received unix signal","sig":"Interrupt"},"target":"watchexec::signal"}
{"timestamp":"2023-11-26T21:27:46.326508Z","level":"INFO","fields":{"message":"action constructed","action":"Action { events: [Event { tags: [Source(Keyboard), Signal(Interrupt)], metadata: {} }], outcome: OnceCell(Uninit) }"},"target":"watchexec::action::worker"}
{"timestamp":"2023-11-26T21:27:46.327687Z","level":"DEBUG","fields":{"message":"running action handler"},"target":"watchexec::action::worker"}
{"timestamp":"2023-11-26T21:27:46.329010Z","level":"DEBUG","fields":{"message":"action handler finished","outcome":"Both(Stop, Exit)"},"target":"watchexec::action::worker"}
{"timestamp":"2023-11-26T21:27:46.330302Z","level":"INFO","fields":{"message":"outcome resolved","outcome":"Both(Stop, Exit)"},"target":"watchexec::action::worker"}
{"timestamp":"2023-11-26T21:27:46.331359Z","level":"DEBUG","fields":{"message":"spawning outcome worker","outcome":"Both(Stop, Exit)","gen":"3"},"target":"watchexec::action::outcome_worker"}
{"timestamp":"2023-11-26T21:27:46.332380Z","level":"DEBUG","fields":{"message":"action process done"},"target":"watchexec::action::worker"}
{"timestamp":"2023-11-26T21:27:46.333909Z","level":"INFO","fields":{"message":"propagating graceful exit","gen":"3"},"target":"watchexec::action::outcome_worker"}
{"timestamp":"2023-11-26T21:27:46.334992Z","level":"DEBUG","fields":{"message":"finishing subtask","subtask":"error_hook"},"target":"watchexec::watchexec"}
{"timestamp":"2023-11-26T21:27:46.336092Z","level":"DEBUG","fields":{"message":"aborting subtask","subtask":"error_hook"},"target":"watchexec::watchexec"}
{"timestamp":"2023-11-26T21:27:46.337121Z","level":"DEBUG","fields":{"message":"aborting subtask","subtask":"action"},"target":"watchexec::watchexec"}
{"timestamp":"2023-11-26T21:27:46.337989Z","level":"DEBUG","fields":{"message":"aborting subtask","subtask":"fs"},"target":"watchexec::watchexec"}
{"timestamp":"2023-11-26T21:27:46.338734Z","level":"DEBUG","fields":{"message":"aborting subtask","subtask":"signal"},"target":"watchexec::watchexec"}
{"timestamp":"2023-11-26T21:27:46.339486Z","level":"DEBUG","fields":{"message":"aborting subtask","subtask":"keyboard"},"target":"watchexec::watchexec"}
{"timestamp":"2023-11-26T21:27:46.340221Z","level":"DEBUG","fields":{"message":"main task graceful exit"},"target":"watchexec::watchexec"}
{"timestamp":"2023-11-26T21:27:46.341304Z","level":"INFO","fields":{"message":"done with main loop"},"target":"watchexec_cli"}
```
- A sample command that you've run that has the issue
`watchexec -vvv --log-file -- python3 inside_test/test.py`

**Comment by passcod** [comment]
Can you replicate with another inotify-based tool in wsl2? Try eg inotifywait. If so, this is a wsl2 bug.

**Comment by passcod** [comment]
You can probably run watchexec on the windows side where it will handle ntfs fine and either call wsl via the command or change a single file and watch _that_ within wsl2.

**Comment by nopeless** [comment]
leaving this here because it is relevant. 

`grep -qEi "(Microsoft|WSL)" /proc/version` - check if OS is wsl

`[[ $(df -T . | awk 'NR==2 {print $2}') == "9p" ]]` - check if current directory uses plan 9 protocol

@chinise also I was referring to user experience, as in trying to use this tool without --poll would show a warning
